one-ui/src/main/java/com/marklogic/hub/curation/controllers/FlowController.java

```diff
@@ -163,4 +163,13 @@ protected ArtifactService getArtifactService() {
         return ArtifactService.on(dataServicesClient);
     }
 
+    private FlowManagerService getFlowManagerService() {
+        lastFlowManagerService = new FlowManagerService(this.hubConfig.getHubConfigImpl());
```

Comment:
I think this introduces a race condition between two threads hitting this controller. Both threads could hit line 167 at the same time, and then lastFlowManagerService would be equal to whichever thread finished 167 last, and both threads would use the same one. 

Is "lastFlowManagerService" only needed for unit testing purposes? If so, the way to get a reference to it is to subclass FlowController in your test and override this method to do the same thing - i.e. hang onto a reference of the FlowManagerService that is constructed.